common/djangoapps/student/helpers.py

```diff
@@ -249,6 +249,7 @@ def get_next_url_for_login_page(request):
         request_params=request_params,
         request_is_https=request.is_secure(),
     )
+    lms_root_url = configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
```

Comment:
Make it a generic variable instead of specific to LMS and use it for the studio root URL as well if the `ROOT_URLCONF` is from CMS and then simply include the `root_url` in `redirect_to` on the basis of `include_host` just before the return. In this way you don't need to do this on multiple places.